src/net.cpp

```diff
@@ -2186,9 +2186,19 @@ CConnman::CConnman(uint64_t nSeed0In, uint64_t nSeed1In) : nSeed0(nSeed0In), nSe
     flagInterruptMsgProc = false;
 }
 
-NodeId CConnman::GetNewNodeId()
+NodeId CConnman::ReserveNewNodeId()
 {
-    return nLastNodeId.fetch_add(1, std::memory_order_relaxed);
+    NodeId newId;
+    {
+        LOCK(cs_setTakenNodeIds);
+        do {
+            if (nLastNodeId == std::numeric_limits<NodeId>::max()) {
+                nLastNodeId = 0;
```

Comment:
Not -1? (or =0 and having the ++ in an else?) Why should it fail to reconsider reusing 0?